circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/replica/hive/DropTableService.java

```diff
@@ -56,13 +100,52 @@ public void removeTableParamsAndDrop(
       }
       client.alter_table(databaseName, tableName, table);
     }
-    LOG
-        .info("Dropping table '{}.{}'.", table.getDbName(), table.getTableName());
-    client.dropTable(table.getDbName(), table.getTableName(), false, true);
+    LOG.info("Dropping table '{}.{}'.", databaseName, tableName);
+    client.dropTable(databaseName, tableName, false, true);
+  }
+
+  private void deleteData(DataManipulationClient dataManipulationClient, String replicaDataLocation) {
+    try {
+      LOG.info("Dropping table data from location: {}.", replicaDataLocation);
+      boolean dataDeleted = dataManipulationClient.delete(replicaDataLocation);
+      LOG.info("Data deleted: {}.", dataDeleted);
+    } catch (IOException e) {
+      LOG.info("Could not drop replica table data at location:{}.", replicaDataLocation);
+    }
+  }
+
+  private void deletePartitionData(
+      DataManipulationClient dataManipulationClient,
+      String replicaTableLocation,
+      List<String> replicaPartitionLocations) {
+    try {
+      LOG.info("Dropping partition data from base location: {}.", replicaTableLocation);
+      for (String location : replicaPartitionLocations) {
+        boolean deleted = dataManipulationClient.delete(location);
+        LOG.debug("Attempted to delete data from location: {}. Successful deletion = {}.", location, deleted);
+      }
+    } catch (IOException e) {
+      LOG.info("Could not drop replica partition data at location:{}.", replicaTableLocation);
+    } catch (UnsupportedOperationException e) {
+      LOG.info(e.getMessage());
+    }
   }
 
   private boolean isExternal(Map<String, String> tableParameters) {
     CaseInsensitiveMap caseInsensitiveParams = new CaseInsensitiveMap(tableParameters);
     return IS_EXTERNAL.equalsIgnoreCase((String) caseInsensitiveParams.get(EXTERNAL_KEY));
   }
+
+  private List<String> getPartitionLocations(CloseableMetaStoreClient client, Table table) {
+    List<String> locations = new ArrayList<>();
+    try {
+      PartitionIterator partitionIterator = new PartitionIterator(client, table, BATCH_SIZE);
+      while (partitionIterator.hasNext()) {
+        locations.add(partitionIterator.next().getSd().getLocation());
```

Comment:
Ah OK, so we do use the iterator but we then add them all to a list so this could still OOM. I think it's alright to do this for now but the "proper" way would be to use the iterator, not an intermediate collection.

Reply:
let's fix this please, we've hit this in the past, it's better to make this resistant now since we're testing it actively and actually have it fresh in our heads.